### DISCUSSION TITLE
Where are syslog event files located?

### DISCUSSION THREAD

**Discussion by sen7inel** [Security-Onion-Solutions/securityonion#6832] [Unsupported Versions] [answered]
Hello all, I'm a new Security Onion user and I need a little help extracting syslog events from my SO server:  I've stood up Security Onion and pointed some Cisco equipment at the server.  I can see the Cisco log event entries in the Elastic web console under dataset syslog.  However, I am unable to locate the syslog files themselves.  Other SO logs are located under /opt/so/log, but I don't see a syslog folder there.  Where are the syslog events stored?

**Comment by weslambert** [comment]
Syslog data that is ingest into Security Onion is located in Elasticsearch indices. Native system syslog data would be in `/var/log/syslog`.   Try running `so-elasticseach-query _cat/indices` and you should see `so-syslog-*` indices (assuming you are running on a search node, or on a clustered node).  As far as extracting them, to what kind of format are you looking to export them? Kibana should allow you to export as CSV if that works. Otherwise, you could use something like an elasticsearch dump script, snapshot the indices using Curator, or just use CLI tools to query and write to a file.

**Comment by sen7inel** [comment, marked as answer]
Thank you for your response and pointing me in the right directon, weslambert.  You were right, running 

PUT so-elastisearch-query
GET _cat/indices 

I can see so-syslog indices.  But how would one go about exporting this to a file, along with the body for the syslog entries.  Kibana does allow to export as a CSV (which works just fine in my case, anything plain text will do), but this only contains the timestamp and id of the event, not the body.  Thank you again for your help, this is still very new to me.

**Reply by tomjones1977** [reply]
You can export a saved search to CSV.  Open the saved search and click on `Share` in the top right corner.  One option there is to export to csv which includes all fields in each record found by the saved search:

![image](https://user-images.githubusercontent.com/32241793/149319522-a3bd36d7-469e-4fa8-a08d-917d6cc5cc53.png)

**Reply by sen7inel** [reply]
Just wanted to follow up - much appreciate the help!  I initially ran into an issue where CSV reports wasn't available from the share menu, and the 'save to csv' option at the top right of my dashboard just exported the default fields.  But that was because I failed to run an actual search.  Once there, I was able to customize my data and retrieve a CSV report as described above.  Thank you again!
